govcd/security_tags_test.go

```diff
@@ -0,0 +1,130 @@
+//go:build network || nsxt || functional || openapi || ALL
+// +build network nsxt functional openapi ALL
+
+package govcd
+
+import (
+	"fmt"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	. "gopkg.in/check.v1"
+	"strings"
+)
+
+func (vcd *TestVCD) Test_SecurityTags(check *C) {
+	skipNoNsxtConfiguration(vcd, check)
+	skipOpenApiEndpointTest(vcd, check, types.OpenApiPathVersion1_0_0+types.OpenApiEndpointSecurityTags)
+
+	securityTagName1 := strings.ToLower(fmt.Sprintf("%s_%d", check.TestName(), 1)) // Security tags are always lowercase in server-side
+	securityTagName2 := strings.ToLower(fmt.Sprintf("%s_%d", check.TestName(), 2))
+
+	// Get testing VM ID
+	testingVM, _ := vcd.findFirstVm(*vcd.vapp)
+	vm := &VM{
+		VM:     &testingVM,
+		client: vcd.org.client,
+	}
+
+	// Create a security tag using UpdateSecurityTag
+	sentSecurityTag := &types.SecurityTag{
+		Tag:      securityTagName1,
+		Entities: []string{testingVM.ID},
+	}
+
+	receivedSecurityTag, err := vcd.client.UpdateSecurityTag(sentSecurityTag)
+	check.Assert(err, IsNil)
+	check.Assert(sentSecurityTag, DeepEquals, receivedSecurityTag)
+
+	// Create a security tag using UpdateVMSecurityTags
+	inputEntitySecurityTags := &types.EntitySecurityTags{
+		Tags: []string{
+			securityTagName1,
+			securityTagName2,
+		},
+	}
+	outputEntitySecurityTags, err := vm.UpdateVMSecurityTags(inputEntitySecurityTags)
+	check.Assert(err, IsNil)
+	check.Assert(outputEntitySecurityTags, NotNil)
+	check.Assert(outputEntitySecurityTags, DeepEquals, inputEntitySecurityTags)
+
+	// Check that the VM with security tags is retrieved using GetSecurityTaggedEntities
+	securityTaggedEntities, err := vcd.client.GetSecurityTaggedEntities("")
```

Comment:
can't we use filter here?

Reply:
The main purpose here is to see that the machine is retrieved by GetSecurityTaggedEntities() since it has been tagged. It doesn't make sense to me here to filter by tag name.